src/Preflight/Preflight.php

```diff
@@ -283,48 +281,48 @@ public function preflight($argv): bool
-            $this->configLocator->addAliasConfig($selfSiteAlias->exportConfig());
-
-            // Process the selected alias. This might change the selected site,
-            // so we will add new site-wide config location for the new root.
-            $root = $this->setSelectedSite($selfSiteAlias->localRoot(), false, $root);
-        }
-
-        // Now that we have our final Drupal root, check to see if there is
-        // a site-local Drush. If there is, we will redispatch to it.
-        // NOTE: termination handlers have not been set yet, so it is okay
-        // to exit early without taking special action.
-        $status = RedispatchToSiteLocal::redispatchIfSiteLocalDrush($argv, $root, $this->environment->vendorPath(), $this->logger());
-        if ($status) {
-            return $status;
-        }
-
-        // If the site locator couldn't find a local site, and we did not
-        // redispatch to a site-local Drush, then we cannot continue.
-        // This can happen when using Drush 9 to call a site-local Drush 8
-        // using an alias record that is only defined in a Drush 8 format.
+        // Note that PreflightSiteLocator::findSite only returns 'false'
+        // when preflightArgs->alias() returns an alias name. In all other
+        // instances we will get an alias record, even if it is only a
+        // placeholder 'self' with the root holding the cwd.
         if (!$selfSiteAlias) {
-            // Note that PreflightSiteLocator::findSite only returns 'false'
-            // when preflightArgs->alias() returns an alias name. In all other
-            // instances we will get an alias record, even if it is only a
-            // placeholder 'self' with the root holding the cwd.
             $aliasName = $this->preflightArgs->alias();
             throw new \Exception("The alias $aliasName could not be found.");
         }
 
-        // If we did not redispatch, then add the site-wide config for the
-        // new root (if the root did in fact change) and continue.
-        $this->configLocator->addSitewideConfig($root);
+        // Record the self alias that we just built or loaded, and apply
+        // any configuration it might contain.
+        $this->aliasManager->setSelf($selfSiteAlias);
+        $this->configLocator->addAliasConfig($selfSiteAlias->exportConfig());
+
+        // Check to see if the alias on the command line points at
+        // a local Drupal site that is not the site at $root
+        $localAliasDrupalFinder = new DrupalFinder();
+        $foundAlternateRoot = $localAliasDrupalFinder->locateRoot($selfSiteAlias->localRoot());
+        if ($foundAlternateRoot) {
+            $alteredRoot = $localAliasDrupalFinder->getDrupalRoot();
+
+            // Now that we have our final Drupal root, check to see if there is
+            // a site-local Drush. If there is, we will redispatch to it.
+            // NOTE: termination handlers have not been set yet, so it is okay
```

Comment:
Can we get rid of ther Drush termination handlers (in a separate issue). I refer to the unusual exit code checking.

Reply:
Maybe; need to check and see what role they are still playing. Would be simpler to not have them.